extension/src/lib/zoom.js

```diff
@@ -65,27 +65,35 @@ fluid.defaults("gpii.chrome.zoom", {
             args: "{arguments}.2"
         },
         "onWindowFocusChanged.applyZoomSettings": "{that}.applyZoomSettings"
+    },
+    members: {
+        magnifierChanged: false
     }
 });
 
 gpii.chrome.zoom.applyZoomInTab = function (that, tab, value) {
-    // set the zoom value if it hasn't already been set.
-    chrome.tabs.getZoom(tab.id, function (currentZoom) {
-        if (currentZoom !== value) {
-            chrome.tabs.setZoom(tab.id, value, function () {
-                if (chrome.runtime.lastError) {
-                    fluid.log("Could not apply zoom in tab'",
-                              tab.url, "', error was: ",
-                              chrome.runtime.lastError.message);
-                    that.events.onError.fire(chrome.runtime.lastError);
-                }
-            });
-        }
-    });
+    if (that.magnifierChanged) {
+        // set the zoom value if it hasn't already been set.
+        chrome.tabs.getZoom(tab.id, function (currentZoom) {
+            if (currentZoom !== value) {
+                chrome.tabs.setZoom(tab.id, value, function () {
+                    if (chrome.runtime.lastError) {
+                        fluid.log("Could not apply zoom in tab'",
+                            tab.url, "', error was: ",
+                            chrome.runtime.lastError.message);
+                        that.events.onError.fire(chrome.runtime.lastError);
+                    }
+                });
+            }
+        });
+    }
 };
 
 gpii.chrome.zoom.applyZoomSettings = function (that) {
     var value = that.model.magnifierEnabled ? that.model.magnification : 1;
+    if (value !== 1) {
+        that.magnifierChanged = true;
+    }
```

Comment:
It seems that this will prevent a previously zoomed page from resetting if the users preference is set to 1.